model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserLoginFailureProvider.java

```diff
@@ -145,4 +152,57 @@ InfinispanChangelogBasedTransaction<LoginFailureKey, LoginFailureEntity> getLogi
     public void close() {
 
     }
+
+    @Override
+    public void migrate(String modelVersion) {
+        if ("26.5.0".equals(modelVersion)) {
+            // This version introduced updated lifetimes for login failures. Recalculate values for existing entries.
+            Cache<LoginFailureKey, SessionEntityWrapper<LoginFailureEntity>> cache = loginFailuresTx.getCache();
+            AggregateCompletionStage<Void> stage = CompletionStages.aggregateCompletionStage();
+            cache.getAdvancedCache().entrySet()
+                    .forEach(entry -> {
+                        RealmModel realm = session.realms().getRealm(entry.getKey().realmId());
+                        if (!realm.isBruteForceProtected()) {
+                            stage.dependsOn(removeKeyFromCache(cache, entry.getKey()));
+                        } else {
+                            updateLifetimeOfCacheEntry(entry, realm, stage, cache);
+                        }
+                    });
+            CompletionStages.join(stage.freeze());
+        }
+    }
+
+    @Override
+    public void updateWithLatestRealmSettings(RealmModel realm) {
+        Cache<LoginFailureKey, SessionEntityWrapper<LoginFailureEntity>> cache = loginFailuresTx.getCache();
+        AggregateCompletionStage<Void> stage = CompletionStages.aggregateCompletionStage();
+        if (!realm.isBruteForceProtected()) {
+            cache.getAdvancedCache().entrySet().stream()
+                    .filter(entry -> Objects.equals(entry.getKey().realmId(), realm.getId()))
+                    .forEach(entry -> stage.dependsOn(removeKeyFromCache(cache, entry.getKey())));
+        } else {
+            cache.getAdvancedCache().entrySet().stream()
```

Comment:
Will brought to my attention that this pattern will trigger the operations for all keys really quickly, which may overload the server. Use instead:

```java
CompletionStages.performConcurrently(<stream here>, <number of concurrent ops / batch size>, Scheduler.from(new WithinThreadExecutor()), <consumer here>);
```

See comment here:
https://github.com/infinispan/infinispan/issues/16245#issuecomment-3577850168